pwa/components/repairers/registration/RegistrationTunnelWorkshop.tsx

```diff
@@ -0,0 +1,466 @@
+
+  const handleCityChange = async (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): Promise<void> => {
+    setCityInput(event.target.value);
+  };
+
+  const handleItinerantCityChange = (value: string): void => {
+    if (value === '' || value.length < 3) {
+      setCitiesList([]);
+    } else fetchCitiesResult(value);
+  };
+
+  const handleCitySelect = (
+    event: SyntheticEvent<Element, Event>,
+    cities: any
+  ) => {
+    let newRepairerCities: RepairerCity[] = [];
+
+    cities.map((city: RepairerCity | City) => {
+      if ('lat' in city && 'lon' in city) {
+        newRepairerCities.push({
+          formatted_name: `${city.name} (${city.postcode})`,
+          latitude: city.lat.toString(),
+          longitude: city.lon.toString(),
+          name: city.name,
+          postcode: city.postcode,
+        } as RepairerCity);
+      } else {
+        newRepairerCities.push(city as RepairerCity);
+      }
+    });
+
+    setRepairerCities(newRepairerCities);
+  };
+
+  const handleChangeStreetNumber = (
+    event: ChangeEvent<HTMLInputElement>
+  ): void => {
+    setStreetNumber(event.target.value);
+  };
+
+  const handleChangeStreet = async (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): Promise<void> => {
+    const adresseSearch = event.target.value;
+    if (adresseSearch.length >= 3) {
+      const streetApiResponse = await searchStreet(adresseSearch, city);
+      setStreetList(streetApiResponse);
+    }
+  };
+  const handleChangeComments = (event: ChangeEvent<HTMLInputElement>): void => {
+    setComment(event.target.value);
+  };
+
+  const handleChangeRepairerType = (
+    event: SelectChangeEvent<typeof repairerTypeSelected>
+  ): void => {
+    const {
+      target: {value},
+    } = event;
+    setRepairerTypeSelected(
+      typeof value === 'string' ? value.split(',') : value
+    );
+  };
+
+  useEffect(() => {
+    if (repairerTypeSelected.length > 0) {
+      setIsRoving(
+        repairerTypeSelected.some(
+          (element) => element === 'Réparateur itinérant'
+        )
+      );
+    }
+  }, [repairerTypeSelected]);
+
+  const handleChangeBikeRepaired = (
+    event: SelectChangeEvent<typeof selectedBikeTypes>
+  ) => {
+    const {
+      target: {value},
+    } = event;
+    setSelectedBikeTypes(typeof value === 'string' ? value.split(',') : value);
+  };
+
+  const handleGoBack = () => {
+    setStepTwoFirstQuestionCompleted(false);
+    router.push('/reparateur/inscription/choix-antenne');
+  };
+
+  const handleNextStep = () => {
+    setStepTwoCompleted(true);
+    router.push('/reparateur/inscription/validation');
+  };
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (!stepTwoFirstQuestionCompleted) {
+        stepOneCompleted
+          ? router.push('/reparateur/inscription/choix-antenne')
+          : router.push('/reparateur/inscription');
+      }
+    }, 100); // Délai de 100 ms
```

Comment:
```suggestion
 useEffect(() => {
    if (!stepOneCompleted || !stepTwoFirstQuestionCompleted) {
      router.push('/reparateur/inscription');
    }
  }, [router, stepOneCompleted, stepTwoFirstQuestionCompleted]);
```
Because if one of this step is false, generally the user come from the browser url so redirect him to the first form step. If a completed step is false but data are in the context, issue come from the code so it is better to check again the form from the begining.